apis/console/v1/policy_types.go

```diff
@@ -0,0 +1,140 @@
+package v1
+
+// +kubebuilder:validation:Optional
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PolicySpec defines Pomerium Enterprise Console namespace
+type PolicySpec struct {
+	// RefID allows to refer to an existing policy created in the console.
+	// Mutually exclusive with any other options.
+	RefID *string `json:"refId,omitempty"`
+
+	// Enforced policy automatically applied to all routes in this and subordinate namespaces.
+	Enforced *bool `json:"enforced,omitempty"`
+
+	Description *string `json:"description,omitempty"`
+	Explanation *string `json:"explanation,omitempty"`
+	Remediation *string `json:"remediation,omitempty"`
+
+	// PPL allows to express the policy in Pomerium Policy Language.
+	// Mutually exclusive with Rego.
+	PPL []PolicyRule `json:"ppl,omitempty"`
+
+	// Rego allows to write raw Rego.
+	// Mutually exclusive with PPL.
+	Rego *string `json:"rego,omitempty"`
+}
+
+// PolicyRule should be either allow or deny.
+type PolicyRule struct {
+	Allow *PolicyAction `json:"allow,omitempty"`
+	Deny  *PolicyAction `json:"deny,omitempty"`
+}
+
+// PolicyAction defines blocks of criteria.
+// Only one operator may be defined.
+type PolicyAction struct {
+	And []PolicyCriteria `json:"and,omitempty"`
+	Or  []PolicyCriteria `json:"or,omitempty"`
+	Not []PolicyCriteria `json:"not,omitempty"`
+	Nor []PolicyCriteria `json:"nor,omitempty"`
+}
+
+// PolicyCriteria specifies PPL matching criteria.
+// Only one criteria may be used.
+// see https://www.pomerium.com/docs/topics/ppl for details.
+type PolicyCriteria struct {
```

Comment:
This is not going to be fun to maintain going forward. It was supposed to be an open ended format.

Reply:
we may set PPL to https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured